Code/Framework/AzCore/AzCore/Debug/PerformanceCollector.cpp

```diff
@@ -0,0 +1,260 @@
+        : m_logCategory(logCategory), m_onBatchCompleteCallback(onBatchCompleteCallback)
+    {
+        for (const auto& metricName : m_metricNames)
+        {
+            [[maybe_unused]] const auto statisticPtr = m_statisticsManager.AddStatistic(metricName, metricName, "us");
+            AZ_Assert(statisticPtr, "Failed to add metric with name <%.*s>. Maybe already added?", AZ_STRING_ARG(metricName));
+        }
+        RestartPeriodicEventStamps();
+    }
+
+    PerformanceCollector::~PerformanceCollector()
+    {
+
+    }
+
+    bool PerformanceCollector::IsWaitingBeforeCapture()
+    {
+        return !m_numberOfCaptureBatches || m_isWaitingBeforeNextBatch;
+    }
+
+    void PerformanceCollector::FrameTick()
+    {
+        if (!m_numberOfCaptureBatches)
+        {
+            return;
+        }
+
+        if (m_isWaitingBeforeNextBatch)
+        {
+            if (m_waitTimeBeforeEachBatch.count() > 0)
+            {
+                AZStd::chrono::steady_clock::time_point now = AZStd::chrono::steady_clock::now();
+                if (m_startWaitTime.time_since_epoch().count() == 0)
+                {
+                    AZ_TracePrintf(LogName, "Will Wait %lld seconds before starting batch %u...", m_waitTimeBeforeEachBatch.count(), m_numberOfCaptureBatches);
+                    m_startWaitTime = now;
+                }
+                auto duration = AZStd::chrono::duration_cast<AZStd::chrono::seconds>(now - m_startWaitTime);
+                if (duration < m_waitTimeBeforeEachBatch)
+                {
+                    return; // Do nothing. Keep waiting.
+                }
+            }
+            AZ_TracePrintf(LogName, "Waited %lld seconds. Will start collecting performance numbers for %u frames at batch %u...", m_waitTimeBeforeEachBatch.count(), m_frameCountPerCaptureBatch, m_numberOfCaptureBatches);
+            m_isWaitingBeforeNextBatch = false;
+        }
+
+        m_frameCountInCurrentBatch++;
+        if (m_frameCountInCurrentBatch <= m_frameCountPerCaptureBatch)
+        {
+            // All good nothing to do.
+            return;
+        }
+
+        // Current batch is complete.
+        m_numberOfCaptureBatches--;
+        m_isWaitingBeforeNextBatch = true;
+        m_frameCountInCurrentBatch = 0;
+        m_startWaitTime = {};
+
+        if (m_dataLogType == DataLogType::LogStatistics)
+        {
+            //It is time to write the statistical summaries to the Log file.
+            RecordStatistics();
+            m_statisticsManager.ResetAllStatistics();
+        }
+        RestartPeriodicEventStamps();
+
+        m_onBatchCompleteCallback(m_numberOfCaptureBatches);
+        if (!m_numberOfCaptureBatches)
+        {
+            // This will close the file that contains performance results for all batches.
+            m_eventLogger.ResetStream(nullptr);
+            AZ_TracePrintf(LogName, "Performance data output file <%s> is ready", m_outputFilePath.c_str());
```

Comment:
Repeat: May want to remove this traceprintf or disable them. Also this is missing a newline as well.
`AZ_TracePrintf` doesn't automatically add a newline unlike AZ_Error/AZ_Warning/AZ_Assert

Reply:
Added new line. Low frequency printf.